scripts/generate_errors.pl

```diff
@@ -203,6 +214,11 @@
                               if ($include_name ne "");
         }
         ${$code_check} .= "\n";
+
+        if (grep(/^$include_name$/, @private_header_files)) {
```

Comment:
Hard-coding a list of which headers are private is not practical for two reasons:

1. At the time we start making the headers private, `generate_errors.pl` needs to work both with the old location and the new location.
2. Later, we'll want to move private headers around or remove some of them, and we don't want to have to change the repository mbedtls at that time.

So instead, please check where the header is located. Something like this (untested):

```suggestion
        # Some crypto headers are "private", meaning they are in the
        # mbedtls/private subdirectory.
        # In case both exist, we'll include the private header, which we'd
        # expect to include the public header.
        if (-e "$crypto_include_dir/mbedtls/private/*.h") {
```